### ISSUE TITLE
When Babel cache gets too big, an exception "TypeError: Method Uint8Array.length called on incompatible receiver" is thrown from `fs.writeFileSync`

### ISSUE THREAD

**Issue by sompylasar** [babel/babel#5259]
This issue: https://github.com/babel/babel/issues/2678 (Babel cache errors when it gets too big)
was improperly fixed via this commit: https://github.com/babel/babel/commit/8ffc7012ccb00e6ab615c55d9911841efb54154a
which worked around the original error but introduced a different error:
```
TypeError: Method Uint8Array.length called on incompatible receiver [object Object]
    at Buffer.get length (native)
    at Object.fs.writeFileSync (fs.js:1161:20)
    at save (/__CENSORED__/node_modules/babel-register/lib/cache.js:45:16)
    at nextTickCallbackWith0Args (node.js:436:9)
    at process._tickDomainCallback (node.js:406:13)
```
which gets thrown from `fs.writeFileSync`: https://github.com/nodejs/node/blob/v4.7.0/lib/fs.js#L1161

The error is caused by erroneously passing an `Object` (`{}`) instead of a `String`, `Buffer` or `UInt8Array` into `fs.writeFileSync`. https://nodejs.org/api/fs.html#fs_fs_writefilesync_file_data_options

The error as of now is in the latest `master` of `babel/packages/babel-register`:
https://github.com/babel/babel/blob/9c2794dc85af9ed1aa6bd319ba279d04fb34e392/packages/babel-register/src/cache.js#L26
```js
/**
 * Write stringified cache to disk.
 */

export function save() {
  let serialised = {};  //<<<<< `serialized` is initialized with a non-serialized `{}`, i.e. `Object`.
  try {
    serialised = JSON.stringify(data, null, "  ");
  } catch (err) {
    if (err.message === "Invalid string length") {
      err.message = "Cache too large so it's been cleared.";
      console.error(err.stack);
      //<<<<< No control flow interruption happens here, fall through, `serialized === {}`.
    } else {
      throw err;
    }
  }
  mkdirpSync(path.dirname(FILENAME));
  fs.writeFileSync(FILENAME, serialised);  //<<<<< `{}` in `serialized` got to here.
}
```

Proposed change:
```diff
  export function save() {
-   let serialised = {};
+   let serialised = "{}";
```

**Comment by xtuc** [comment]
Good catch :+1: This would have catch using Flow annotations I think. I will make a PR for that.

**Comment by xtuc** [comment]
Closed in #5260. Feel free to reopen this issue.
